Koha/Illrequest.pm

```diff
@@ -1346,57 +1330,282 @@ EOF
             "No target email addresses found. Either select at least one partner or check your ILL partner library records.")
           if ( !$to );
         # Create the from, replyto and sender headers
-        my $from = $branch->branchemail;
+        my $from = $branch->branchillemail || $branch->branchemail;
         my $replyto = $branch->branchreplyto || $from;
         Koha::Exceptions::Ill::NoLibraryEmail->throw(
             "Your library has no usable email address. Please set it.")
           if ( !$from );
 
-        # Create the email
-        my $message = Koha::Email->new;
-        my %mail = $message->create_message_headers(
-            {
-                to          => $to,
-                from        => $from,
-                replyto     => $replyto,
-                subject     => Encode::encode( "utf8", $params->{subject} ),
-                message     => Encode::encode( "utf8", $params->{body} ),
-                contenttype => 'text/plain',
+        # So we get a notice hashref, then substitute the possibly
+        # modified title and body from the draft stage
+        my $letter = $self->get_notice({
+            notice_code => 'ILL_PARTNER_REQ',
+            transport   => 'email'
+        });
+        $letter->{title} = $params->{subject};
+        $letter->{content} = $params->{body};
+
+        # Send the email
+        my $params = {
+            letter                 => $letter,
+            borrowernumber         => $self->borrowernumber,
+            message_transport_type => 'email',
+            to_address             => $to,
+            from_address           => $from
```

Comment:
I don't see the $replyto variable from above used at all here now.. we should use the 'reply_address' param introduced with bug 10269 here populated with the above variable.